### PR TITLE
Implement `--dry-run` and `--deploy-app` flags for `deploy.sh`

### PR DESCRIPTION
**Pull request by dhaiducek**
- `deploy.sh` outputs the resulting Subscription manifests to a file every time, regardless of whether in dry-run mode or not
- `--dry-run` - Output the result to `stdout` but don't apply it to the cluster (also saves the result to `manifests.yaml`)
- `--deploy-app` - Deploy an Application manifest for additional visibility in the UI (also updates the Subscription to use Placement since the UI doesn't support the full topology otherwise)
- Updates to the various `README.md` for these changes as well as clarifications around the generator and PolicySets

### CODE REVIEW

File: deploy/deploy.sh [hunk 1]
```diff
@@ -98,6 +110,20 @@ while read -r -p "Would you like to proceed (y/n)? " response; do
   esac
 done
 
+# Check for the namespace
```
Comment (gparvin) [inline deploy/deploy.sh]: Should the namespace be created in a dry run scenario?

**Review by gparvin — CHANGES_REQUESTED** [review]

File: deploy/deploy.sh (hunk 1 above)
Comment (dhaiducek) [reply deploy/deploy.sh]: Ooo, you're right--good catch!

**Comment by gparvin** [comment]
/lgtm
This looks like great stuff!  Thanks for the updates.

**Comment by dhaiducek** [comment]
/hold I'm not 100% sure we should add the Application feature: https://coreos.slack.com/archives/C027TN14SGJ/p1647613171020129?thread_ts=1647612794.419359&cid=C027TN14SGJ

(Well, at least it looks like we should add a Placement[Rule] if we do go down that road)

**Comment by dhaiducek** [comment]
Okay, PlacementRule added. (Which we'll have to update to Placement eventually, but that might be a task for another day...) Please take another look. The logic flows got a little bit more complex with Placement in the mix, but I'm not sure about alternative tooling (or Kustomize features I may not be using).

**Comment by gparvin** [comment]
Ah, yes, so you want it to look nice in the Topology part of App Lifecycle.  It will be a nice thing to show off in the sync call too.
/lgtm

**Review by gparvin — APPROVED** [review]

**Pull request merged**